provd/internal/services/user/user_test.go

```diff
@@ -298,56 +342,71 @@ func newUserClient(t *testing.T, accountsEndpoint string, hostnameEndpoint strin
 	return pb.NewUserServiceClient(conn)
 }
 
-type accountsdbus struct {
-	endpoint  string
-	wantError bool
+type accountsdbus struct{}
+
+func (a accountsdbus) Ping() *dbus.Error {
+	return nil
 }
 
 func (a accountsdbus) FindUserByName(name string) (string, *dbus.Error) {
-	if a.wantError {
-		return "", dbus.NewError("org.freedesktop.Accounts.Error.Failed", []interface{}{"error"})
+	if name == "error" {
+		return "", dbus.NewError("org.strange.error", []interface{}{"error"})
 	}
-	return "/org/freedesktop/Accounts/Usernoerror", nil
+	// This is used to also include the username in the error log, used to determine its an error because a user wasn't found.
+	if name != "ok" {
+		return "", dbus.NewError("org.freedesktop.Accounts.Error.Failed", []interface{}{"Your name was: " + name})
+	}
+	return "/org/freedesktop/Accounts/Userok", nil
```

Comment:
I hardcoded the returned user because that is the path of the user I create when I export the dbus mocks at the start of TestMain(). So in all the positive cases, they are interacting with the same pre-exported user object mock?

Reply:
Path is actually dependent on the UserID I see. So Creating a second user on my PC with username matt2 will return `/org/freedesktop/Accounts/User1001`